jupyterhub/templates/image-puller/_helpers-daemonset.tpl

```diff
@@ -50,12 +50,11 @@ spec:
       {{- if and (not .hook) .Values.scheduling.podPriority.enabled }}
       priorityClassName: {{ include "jupyterhub.user-placeholder-priority.fullname" . }}
       {{- end }}
-      tolerations:
-        {{- include "jupyterhub.userTolerations" . | nindent 8 }}
-        {{- with .Values.prePuller.extraTolerations  }}
-        {{- . | toYaml | trimSuffix "\n" | nindent 8 }}
-        {{- end }}
       nodeSelector: {{ toJson .Values.singleuser.nodeSelector }}
+      {{- with include "jupyterhub.userTolerations" (dict "root" . "extraTolerations" .Values.prePuller.extraTolerations) }}
+      tolerations:
+        {{- . | nindent 8 }}
+      {{- end }}
```

Comment:
@kafonek you asked about this section before. The hook-image-puller and continuous-image-puller pods part of these daemonsets are considered "user pods" so they should get the user associated tolerations, but they should also have their dedicated additional tolerations if set: prePuller.extraTolerations

The need for being able to do this is described in #1883.